snapcraft/_store.py

```diff
@@ -281,3 +283,92 @@ def download(snap_name, channel, download_path, arch):
         raise RuntimeError(
             'Failed to download {} at {} (mismatched SHA)'.format(
                 snap_name, download_path))
+
+
+def _get_snap_data(name, store):
+    try:
+        details = store.cpi.get_package(name, 'stable', None)
```

Comment:
get_package() is not prepared for receiving arch=None, it's pushing a bogus header for CPI waiting for a disaster to happen ... arch is not important for snap-id lookups, so adjust get_package to suppress 'X-Ubuntu-Architecture' is arch is None (similar tweak on errors.SnapNotFound)
Also, I don't see much value in this wrapper if SnapNotFound message is sane, no need to turn it into RuntimeError, 2 different things.